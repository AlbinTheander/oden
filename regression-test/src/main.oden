;; -*- mode: lisp;-*-

;; package declaration
(pkg main)

;; import declaration
(import fmt)

;; function definition can also use the usual scheme shorthand
(def (identity-string x) (++ "" x))

;; value definition
(def result (++ (identity-string "Hello") ", world!"))

;; main function definition, must have type (-> unit)
(def main (fn () (fmt.Println result)))
