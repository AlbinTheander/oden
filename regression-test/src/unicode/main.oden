(pkg unicode/main)

(import fmt)

(def (main) (fmt.Println "As a Swede, åäö is what I want to write."))
