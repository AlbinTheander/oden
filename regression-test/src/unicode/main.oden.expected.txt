As a [Swede], åäö is what I want to write. \\ oh yeah
